File: Bookify/src/app/feature-modules/reservation/model/ReservationDTO.ts
import {UserReservationDTO} from "./UserReservationDTO";

export interface ReservationDTO {
  id: number;
  created: Date;
  start: string;
  end: string;
  guestNumber: number;
  price: number;
  status: string;
  user: UserReservationDTO;
  accommodationId: number;
  accommodationName: string;
  avgRating: number;
  imageId: number;
}

File: bookify/bookify-server/src/main/java/rs/ac/uns/ftn/Bookify/dto/ReservationDTO.java
package rs.ac.uns.ftn.Bookify.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import rs.ac.uns.ftn.Bookify.enumerations.Status;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ReservationDTO {
    private Long id;
    private LocalDate created;
    private String start;
    private String end;
    private int guestNumber;
    private double price;
    private Status status;
    private UserReservationDTO user;
    private Long accommodationId;
    private String accommodationName;
    private double avgRating;
    private Long imageId;
}
